DatadogRUM/Sources/Instrumentation/AppHangs/AppHangsObserver.swift

```diff
@@ -44,16 +57,30 @@ internal class AppHangsObserver: RUMCommandPublisher {
     }
 
     private func report(appHang: AppHang) {
-        let addHangCommand = RUMAddCurrentViewErrorCommand(
-            time: appHang.date,
-            message: "App Hang",
-            type: "AppHang",
-            stack: nil, // TODO: RUM-2925 Add hang stack trace
-            source: .source,
-            attributes: [
-                "hang_duration": appHang.duration
-            ]
-        )
-        subscriber?.process(command: addHangCommand)
+        var command: RUMAddCurrentViewErrorCommand
+
+        if let backtrace = appHang.backtrace {
+            command = RUMAddCurrentViewErrorCommand(
+                time: appHang.date,
+                message: Constants.appHangErrorMessage,
+                type: Constants.appHangErrorType,
+                backtrace: backtrace,
+                source: .source,
+                attributes: [:]
+            )
+        } else {
+            command = RUMAddCurrentViewErrorCommand(
+                time: appHang.date,
+                message: Constants.appHangErrorMessage,
+                type: Constants.appHangErrorType,
+                stack: Constants.appHangNoStackErrorMessage,
+                source: .source,
+                attributes: [:]
+            )
+        }
+
+        command.attributes["hang_duration"] = appHang.duration
```

Comment:
/suggestion This is the result of wrong `RUMAddCurrentViewErrorCommand.init` definitions, maybe we can revisit and use one like:
```swift
    init(
        time: Date,
        message: String,
        type: String? = nil,
        stack: String? = nil,
        source: RUMInternalErrorSource = .source,
        errorSourceType: RUMErrorEvent.Error.SourceType = .ios,
        isCrash: Bool = false,
        threads: [DDThread]? = nil,
        binaryImages: [BinaryImage]? = nil,
        isStackTraceTruncated: Bool? = nil,
        attributes: [AttributeKey: AttributeValue] = [:]
    ) {
        self.time = time
        self.attributes = attributes
        self.message = message
        self.type = type
        self.stack = stack
        self.isCrash = isCrash
        self.source = source
        self.errorSourceType = errorSourceType
        self.threads = threads
        self.binaryImages = binaryImages
        self.isStackTraceTruncated = isStackTraceTruncated
    }
```

Reply:
TBH I have no idea why I added next initializer. Update to use an existing one 🧽, thanks for bringing this up 👍.